src/variableDiagnosticsProvider.ts

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+import { workspace, languages, Diagnostic, DiagnosticSeverity, DiagnosticCollection, TextDocument, Range, Position } from 'vscode';
+
+import { OnRequestVariableEvent } from "./events/requestVariableEvent";
+import { VariableProcessor } from "./variableProcessor";
+
+export class VariableDiagnosticsProvider {
+    private httpDiagnosticCollection: DiagnosticCollection;
+
+    constructor() {
+        this.httpDiagnosticCollection = languages.createDiagnosticCollection();
+
+        this.checkVariablesInAllTextDocuments();
+
+        OnRequestVariableEvent(this.checkVariablesInAllTextDocuments);
```

Comment:
Sorry, you should still use __arrow function__ syntax, otherwise would cause an incorrect `this` binding when callback `checkVariablesInAllTextDocuments` called.